src/SplitTrilinearInterpolation.jl

```diff
@@ -0,0 +1,174 @@
+module SplitTrilinearInterpolation
+using DocStringExtensions
+
+import ..Interpolation1D: Linear, Flat, interpolate1d!
+import ..BilinearInterpolation: Bilinear, interpolatebilinear!, get_dims
+
+"""
+    SplitTrilinearHV{VS, VT, HB, W1, W2}
+
+This struct stores the source and target grids and work arrays for split trilinear interpolation.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct SplitTrilinearHV{VS, VT, HB, W1, W2}
+    "Vertical source grid"
+    vsource::VS
+    "Vertical target grid"
+    vtarget::VT
+    "bilinear object containing the horizontal source and target grids"
+    bilinear::HB
+    "work array, storing fsource on the target horizontal grid"
+    fsourceintermediate::W1
+    "vertical source grid corresponding to target horizontal grid"
+    vsourceintermediate::W2
+end
+
+"""
+    SplitTrilinearHV(
+        (vsource, h1source, h2source),
+        (vtarget, h1target, h2target);
+    )
+
+A constructor for the `SplitTrilinearHV` struct.
+"""
+function SplitTrilinearHV(
```

Comment:
Could you explain more clearly in the docstring what the inputs to this function are please? Are these spaces, and if so do the dimensions of different spaces have to be the same?